x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -166,7 +166,7 @@ function isLoginAttemptWithProviderType(
  * @param sessionValue The session value to check.
  */
 function isLegacyProviderSession(sessionValue: any) {
-  return typeof sessionValue?.provider === 'string';
+  return typeof sessionValue?.provider?.name !== 'string';
```

Comment:
__optional nit:__ probably more Typescript'y solution would be to reverse check and use type guard instead. But up to you.
```ts
function isSupportedProviderSession(sessionValue: any): sessionValue is ProviderSession {
  return typeof sessionValue?.provider?.name === 'string';
}
```

Reply:
I like it, updated.